src/structs.py

```diff
@@ -128,6 +128,10 @@ def set(self, obj: Object, feature_name: str, feature_val: Any) -> None:
         idx = obj.type.feature_names.index(feature_name)
         self.data[obj][idx] = feature_val
 
+    def get_objects(self, object_type: Type) -> List[Object]:
+        """Return objects of the given type in the order of __iter__()."""
+        return [o for o in self if o.type == object_type]
```

Comment:
This would be useful in a lot of places right? I guess it's too hard to figure out those places though. But we no longer need patterns like `for obj in state: if obj.type == blah: process(obj)`

Reply:
yeah... I'm too lazy to go hunting, do you want to? 😛

Reply:
no